### ISSUE TITLE
Check public APIs for Swift friendliness

### ISSUE THREAD

**Issue by philipphofmann** [getsentry/sentry-cocoa#1844]
### Description

Go through all public APIs and check, if they conform to [Swift naming guidelines](https://www.swift.org/documentation/api-design-guidelines/).

- [ ] https://github.com/getsentry/sentry-cocoa/issues/1842

**Comment by philipphofmann** [comment]
@sindresorhus, we are currently working on this task. As you voted with 👍, are there any APIs you can point out that are not Swift-friendly and that you would like to change?

**Comment by philipphofmann** [comment]
Classes not prefixed with Sentry that we could consider prefixing with Sentry in Swift. We also now prefixed User and Client with Sentry. We renamed `Client` to `SentryClient` because we have `SentrySDK` and `SentryHub` and there could be naming collisions. We renamed `User` to `SentryUser` as almost every application has a user class. 

Does the same apply to `Event`? I guess many apps could also have an `Event` class. Maybe we should also rename it to `SentryEvent`?

The following classes are very specific to Sentry. They could have a prefix `Sentry`, but most users don't interact with them, so the chances of naming collisions in the same Swift file are low. I think we could keep them, cause renaming them causes plenty of hassle in our test code:
* Exception
* Mechanism
* MechanismMeta
* Frame
* DebugMeta

What do you think @brustolin and @kevinrenskers?

**Comment by kevinrenskers** [comment]
To be honest I'm not sure about the User rename. Name collisions are not really a thing in Swift, as you can always namespace it: `Sentry.User`. 

Have we had user complaints about the User class?

The Client rename does make a lot of sense to me.

Exception is very generic but yeah with namespaces it's fine.

**Comment by philipphofmann** [comment]
> To be honest I'm not sure about the User rename. Name collisions are not really a thing in Swift, as you can always namespace it: Sentry.User.

Yes, but user is pretty common. I think that's fine. I would keep the others.

> Have we had user complaints about the User class?

Yes one user, but that shouldn't be the reason why we do it. We should do the right thing.

**Comment by brustolin** [comment]
I think we should change `Event` to `SentryEvent`, `Exception` maybe.
Im fine with the others.

> as you can always namespace it: Sentry.User.

Yeap, but this is annoying.

**Comment by philipphofmann** [comment]
[Java](https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/SentryEvent.java), [Dart](https://github.com/getsentry/sentry-dart/blob/main/dart/lib/src/protocol/sentry_event.dart), and [.NET](https://github.com/getsentry/sentry-dotnet/blob/main/src/Sentry/SentryEvent.cs) also have SentryEvent. [JavaScript](https://github.com/getsentry/sentry-javascript/blob/master/packages/types/src/event.ts) on the other hand has `Event.

I'm also fine with keeping `Event`. It causes plenty of hassle, and what's the benefit? On the other hand, we just renamed client and user, so it would be consistent.

**Comment by kahest** [comment]
Check if Swift supports only prefixing one e.g. `User` class used in a file, e.g. prefix `Sentry.User` and keep instances of other `User` classes as `User`. If possible, then revert renaming of `SentryUser`, otherwise keep to avoid devs needing to prefix their own `User` classes (potentially within existing implementations).

**Comment by philipphofmann** [comment]
@brustolin, you now merged https://github.com/getsentry/sentry-cocoa/pull/2462. So we can close this.
